nova/virt/vmwareapi/vmops.py

```diff
@@ -2221,6 +2222,68 @@ def manage_image_cache(self, context, instances):
             datastores_info.append((ds, dc_info))
         self._imagecache.update(context, instances, datastores_info)
 
+        if CONF.vmware.image_as_template:
+            self._age_cached_image_templates(dc_info, instances)
+
+    def _age_cached_image_templates(self, dc_info, instances):
+        project_ids = set()
+        for inst in instances:
+            project_ids.add(inst.project_id)
+        for project_id in project_ids:
+            folder_ref = self._get_project_folder(dc_info,
+                                                  project_id=project_id,
+                                                  type_='Images')
+            self._destroy_expired_image_templates(self._session, folder_ref)
+
+    def _destroy_expired_image_templates(self, session, templ_vm_folder_ref):
+        client_factory = session.vim.client.factory
+        task_filter_spec = client_factory.create('ns0:TaskFilterSpec')
+        task_filter_spec.entity = client_factory.create(
+                                        'ns0:TaskFilterSpecByEntity')
+        task_filter_spec.entity.entity = templ_vm_folder_ref
+        task_filter_spec.entity.recursion = "children"
+
+        expired_templ_vms = {}
+        try:
+            task_collector = session._call_method(session.vim,
+                                          "CreateCollectorForTasks",
+                                          session.vim.service_content.taskManager,
+                                          filter=task_filter_spec)
+
+            while True:
+                page_tasks = session._call_method(session.vim,
+                                                  "ReadNextTasks",
+                                                  task_collector,
+                                                  maxCount=10)
+                if len(page_tasks) == 0:
+                    break
+
+                for ti in page_tasks:
+                    templ_vm_ref = None
+                    if ti.descriptionId == "VirtualMachine.clone":
+                        templ_vm_ref = ti.entity
+                    elif ti.descriptionId == "Folder.createVm":
```

Comment:
Is this the initial create of the Template?

Reply:
Yes.

Reply:
Could you add this as a comment above the line, please?

Reply:
Changed it to the correct task for template-based approach. And added comment.